pytket/extensions/quantinuum/backends/quantinuum.py

```diff
@@ -372,39 +372,66 @@ def required_predicates(self) -> List[Predicate]:
+    @property
+    def _two_qubit_gate_set(self) -> Set[OpType]:
+        """
+        Assumes that only possibly supported two-qubit gates are
+        ZZPhase, ZZMax and TK2.
+
+        :return: Set of two-qubit OpType in gateset.
+        :rtype: Set[OpType]
         """
-        :param implicit_swaps: If true, allows rebasing of Circuit via TK2 gates
-            to use implicit wire swaps in circuit construction if it reduces
-            the total 2qb qate count.
-        :type implicit_swaps: bool
+        return self._gate_set & set([OpType.ZZPhase, OpType.ZZMax, OpType.TK2])
+
+    def rebase_pass(self, **kwargs: QuumKwargTypes) -> BasePass:
+        """
+        Supported kwargs:
+        * `implicit_swaps`: Boolean flag, which if true, returns
+            rebasing pass that allows implicit wire swaps.
+            Default False.
+        * `target_2qb_gate`: pytket OpType, if provided, will
+            return a rebasing pass that only allows given
+            two-qubit gate type.
         :return: Compilation pass for rebasing circuits
         :rtype: BasePass
         """
-        return auto_rebase_pass(self._gate_set, implicit_swaps)
+        target_2qb_optype: OpType = kwargs.get("target_2qb_gate", OpType.ZZMax)
+        if target_2qb_optype not in self._two_qubit_gate_set:
+            raise QuantinuumAPIError(
+                "Requested target_2qb_gate is not supported by the given Device. "
+                "Please check _two_qubit_gate_set attribute to see which two-qubit "
+                "gates are supported."
+            )
+        return auto_rebase_pass(
+            self._gate_set - self._two_qubit_gate_set | {target_2qb_optype},
+            allow_swaps=bool(kwargs.get("implicit_swap", True)),
+        )
 
-    def default_compilation_pass(
-        self, optimisation_level: int = 2, implicit_swaps: bool = True
+    def default_compilation_pass_with_options(
+        self, optimisation_level: int = 2, **kwargs: QuumKwargTypes
     ) -> BasePass:
         """
         :param optimisation_level: Allows values of 0,1 or 2, with higher values
             prompting more computationally heavy optimising compilation that
             can lead to reduced gate count in circuits.
         :type optimisation_level: int
-        :param implicit_swaps: If true, allows rebasing of Circuit via TK2 gates
-            to use implicit wire swaps in circuit construction if it reduces
-            the total 2qb qate count.
-        :type implicit_swaps: bool
         :return: Compilation pass for compiling circuits to Quantinuum devices
         :rtype: BasePass
+
+        Supported kwargs:
+        * `implicit_swaps`: Boolean flag, which if true, allows rebasing of
+            Circuit via TK2 gates to use implicit wire swaps in circuit
+            construction if it reduces the total 2qb qate count.
+        * `target_2qb_gate`: pytket OpType, if provided, will rebase
```

Comment:
```suggestion
        * `target_2qb_gate`: :py:class:`OpType`, if provided, will rebase
```
Hopefully will generate a hyperlink in the docs.